### ISSUE TITLE
warning undocumented rule def

### ISSUE THREAD

**Issue by Aladdin-ADD** [eslint/eslint#10730]
**The version of ESLint you are using.**
master
**The problem you want to solve.**
I noticed eslint is still allowing rule def like this:
```js
// some-rule.js
module.exports = context => {...}
```
it should be
```js
module.exports = {
  meta: {...},
  create: context => {...}
}
```

and it was used in rule tester, like:
https://github.com/eslint/eslint/blob/master/tests/fixtures/testers/rule-tester/no-var.js#L10-L32

accordingly it might add the complexity to validate rules and cause inconsistency. e.g 
https://github.com/eslint/eslint/blob/137140f737d720fa93a4ac06ce79a86b523327f7/tests/lib/testers/rule-tester.js#L252-L266

should throw an error, since no `meta.fixable`

**Your take on the correct solution to problem.**

though it's not documented, I'm not sure if it's too bold to disallow the usage. hopefully emitting an warning is the first step to the right way.

https://github.com/eslint/eslint/blob/master/lib/rules.js#L51
